addons/mrp_subcontracting/wizard/stock_picking_return.py

```diff
@@ -16,14 +16,14 @@ def _compute_subcontract_location_id(self):
                 record.picking_id.company_id
             ).property_stock_subcontractor
 
-    @api.onchange('picking_id')
-    def _onchange_picking_id(self):
-        res = super(ReturnPicking, self)._onchange_picking_id()
-        if any(return_line.quantity > 0 and return_line.move_id.is_subcontract for return_line in self.product_return_moves):
-            self.location_id = self.picking_id.partner_id.with_company(self.picking_id.company_id).property_stock_subcontractor
+    def _compute_moves_locations(self, vals):
+        res = super(ReturnPicking, self)._moves_locations_from_picking(vals)
+        if any(return_line[2]['quantity'] > 0 and self.env['stock.move'].browse(return_line[2]['move_id']).is_subcontract for return_line in vals['product_return_moves']):
+            picking = self.env['stock.picking'].browse(vals['picking_id'])
+            vals['location_id'] = picking.partner_id.with_company(picking.company_id).property_stock_subcontractor
         return res
 
     def _prepare_move_default_values(self, return_line, new_picking):
-        vals = super(ReturnPicking, self)._prepare_move_default_values(return_line, new_picking)
+        vals = super(ReturnPicking, self)._moves_locations_from_picking(return_line, new_picking)
```

Comment:
and this I believe is a mistake, it should stay `_prepare_move_default_values` :thinking: